mycelial-crdt/src/list.rs

```diff
@@ -30,24 +30,45 @@ use std::collections::BTreeMap;
 
 /// Uniquely identifies operation for a given process
 #[derive(Debug, PartialEq, Eq, PartialOrd, Ord, Clone, Serialize, Deserialize)]
-pub struct Key {
-    id: Ratio<BigInt>,
+pub struct Key<T> {
+    id: T,
     process: u64,
     op: u64,
 }
 
-impl Key {
+impl<T: Clone> Key<T> {
     /// Construct new key for a given process and operation number
-    pub fn new<T: Into<Ratio<BigInt>>>(id: T, process: u64, op: u64) -> Self {
+    pub fn new<R: Into<T>>(id: R, process: u64, op: u64) -> Self {
         Self {
             id: id.into(),
             process,
             op,
         }
     }
+}
+
+/// List Key trait
+pub trait ListKey: Sized + Ord + Eq + Clone {
+    /// Defines behaviour of the list by defining implementation of function `between`
+    fn between(
+        process: u64,
+        op: u64,
+        left: Option<&Self>,
+        right: Option<&Self>,
+    ) -> Result<Self, ListError>;
 
+    /// Check is key is arbitraty precision
```

Comment:
```suggestion
    /// Check if key supports  arbitrary precision arithmetic
    ///
    /// If it does - insert at any position is allowed
    /// Otherwise - it's a key for append-only list
```